### ISSUE TITLE
Editing `UIFormatters` resources in Specify 6 erases 7-only attributes

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3763]
See https://github.com/specify/specify6/issues/1211

7 supports both a `value` and `pattern` for regex formatters, but 6 replaces the `value` with the `pattern` and deletes the `pattern` entirely.

Similar issue to #3654

**Comment by melton-jason** [comment]
> @melton-jason thinks this is best resolved with a Specify 6 update

If we want to maintain compatibility between Specify 6 and Specify 7 Field Formats, this can _only_ be resolved by a Specify 6 update. 
Whenever Specify 6 overwrites the UIFormatters file after making a change in Schema Config, it is setting the `value` of the format equal to the `pattern`--- and completely removes the actual regular expression. 

As far as I know, the value of the format should always be the actual regular expression, and the pattern should be what is displayed to the user when entering data in the field.

Without the regular expression, there is no way for Specify 6 or Specify 7 to validate whether a string is valid to the regular expression, and there is no way for Specify 7 to construct the regular expression from the pattern.

**Comment by maxxxxxdlp** [comment]
FYI, when I looked at sp6 souce code last, after user makes changes, it generates XML from scratch, using only the attributes it is aware of - unlike the form editor in sp7 which preserves all existing unknown attributes.

Sp6's approach is simpler, and does not leave garbage attributes when they are no longer needed, but is fundamentally incompatible with any other system - changing it to look at existing attributes would likely be tricky, so the solution is to retire sp6 or modify it to preserve a few more attributes (from a hardcoded list of options)

**Comment by melton-jason** [comment]
Yes you're correct. Specify 6 uses only the information stored in the JVM memory (or by proxy the database) to write to the local files such as UIFormatters. 
Since Specify 6 never stores the Specify 7 specific information in memory, it does not write it back to the file(s).

**Comment by bronwyncombs** [comment]
**Reported by Corinna @ CSIRO:**

> Not sure whether this is a bug, but I just wanted to share our recent experience with the DataObjFormatter.xml after our upgrade to 7.9.3.1.
> 
> As part of our insect collection data migration to Specify, we set up quite a few formatters and aggregators. I’ve switched between making edits via S6 and S7 quite regularly and never observed any issues. That is, when I made a change in S7 it appeared in S6 and the other way around.
> 
> About 1.5 weeks ago, we cut over to 7.9.3.1. Earlier this week, I went into S6 to make a change to one of the formatters and I noticed that not a single table had an aggregator set in the schema config. This appeared strange, but I went ahead anyway with my edit. When I opened the xml in S7 shortly after, ALL aggregators had disappeared. Luckily, we had the ‘original’ version in one of the other environments which I downloaded and re-applied to our Production environment.
> 
> After a lot of trial and error and investigating, we found that one field had two aggregators defined; both of them set to ‘default=true’. S6 didn’t like this and (I presume) reverted to some default DataObjFormatter.xml that had no aggregators defined which was carried through to S7. Fortunately, I noticed this in time and was able to restore the previous file. After fixing the duplication everything was fine.
> 
> So in short, it appears that S6 is resetting the content of the DataObjFormatter.xml when there’s an error in the xml without any notification to the user. This only seems to be happening in the latest version – the duplication ‘error’ was present in the xml before the upgrade and loaded fine in S6 (ie., all aggregators were present).

**Additional Context**

Unable to recreate on localhost instance using `fwri` database. Specifically, I made two of their CE formatters both `default="true"`, opened schema config in Sp6 and confirmed that tables still had aggregators set. I then opened Sp7 locally again and confirmed that the DataObjFormatter definition was still there and had not reverted to default. I did this process running both `v7.9.3.1` and `edge` with the same results. 

Because this is a closely related issue with app resources and 6/7 compatibility, I thought I'd leave this here instead of opening a new issue since it has not yet been recreated internally.

**Comment by maxpatiiuk** [comment]
xml-editor has the UI for DataObjFormatters, and so sp7 should be used over sp6 for editing them from now on - sp7 never removes attributes it doesn't recognize, thus avoiding this bug
we should have UI for UIFormatters too at some point
tracked in https://github.com/specify/specify7/issues/1156
